packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -642,7 +642,7 @@ abstract class RenderSliverFloatingPinnedPersistentHeader extends RenderSliverFl
       paintExtent: clampedPaintExtent,
       layoutExtent: layoutExtent.clamp(0.0, clampedPaintExtent) as double,
       maxPaintExtent: maxExtent + stretchOffset,
-      maxScrollObstructionExtent: maxExtent,
+      maxScrollObstructionExtent: minExtent,
```

Comment:
Commenting here in case I forget when reviewing the tests: Did you add a test for this?

Reply:
I'll add one.